### PR TITLE
Wide support

### PR DESCRIPTION
**Pull request by WillDenissen**
Implements issue https://github.com/eliben/pyelftools/issues/405 for -W --wide support for readelf.py

### CODE REVIEW

File: scripts/readelf.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def _get_cu_base(cu):
 class ReadElf(object):
     """ display_* methods are used to emit output into the output stream
     """
-    def __init__(self, file, output):
+    def __init__(self, file, output, args):
```
Comment (eliben) [inline scripts/readelf.py:101]: Please add docstring section for the new parameter `args`

File: scripts/readelf.py [hunk 2]
```diff
@@ -251,20 +252,24 @@ def display_program_headers(self, show_heading=True):
             (Elf file type is...)
         """
         self._emitline()
-        if self.elffile.num_segments() == 0:
+        nseg = self.elffile.num_segments()
```
Comment (eliben) [inline scripts/readelf.py:255]: I'd rather you just inline the call everywhere, this isn't much of a performance optimization anyhow

**Review by eliben — COMMENTED** [review]
Quick round. Would be happy if @sevaa took a look at the PR as well

**Comment by sevaa** [comment]
I can only reiterate that I don't see the point of this feature, unless someone wants to repurpose readelf.py as a more or less drop-in replacement for GNU readelf. Which is only needful if one is parsing its output for further programmatic processing, which I can't possibly endorse, since my preferred approach would be writing your CLI based on pyelftools.

From what I can see, autotest against `readelf -W` with GNU is not in the books. That's good, since attempting that would incur a ton of maintenance down the road, as it would cause twice the amount of output format discrepancies.

That said, the core library is intact. Merge away, no hard veto from me.

**Comment by eliben** [comment]
@sevaa makes a good point, and I'm not sure you've addressed that in the original issue, @WillDenissen. Indeed, full feature parity between `readelf.py` and the GNU tool is not a goal of the project - it's a huge amount of work overall and the benefit is unclear. Obviusly, `readelf.py` demonstrates that such a tool _can_ be built using `elftools`, and I encourage others to develop it separately, if desired.

Can you please comment whether this PR improves the coverage of the elftools library itself, or whether it's primarily an extension to the `readelf.py` tool?

**Comment by WillDenissen** [comment]
This PR does not improve the coverage of the elftools library itself. It only aligns the readelf.py output with the readelf reference implementation. It was for me an exercise to implement a "harmless feature" and go through the PR workflow for the first time.

**Comment by sevaa** [comment]
The reference implementation itself is a moving target, too. The version that we have here has spotty DWARFv5 support, and the latest master has gratuitous changes in output format here and there.

**Pull request closed without merging**